### ISSUE TITLE
BadParameterException should be unchecked

### ISSUE THREAD

**Issue by dagnir** [HawaiiStateDigitalArchives/PID-webservice#37]
The exception is thrown when the API user makes a request with invalid input; this should be a case where we use an unchecked exception (extends `RuntimeException`) since there's really nothing sensible the code can do other than tell the user that their input was bad. In general, checked exceptions (extending from `Exception`) are used if we foresee the caller being able to recover from the error.

**Comment by ttaomae** [comment]
I would actually argue that `BadParameterException` can be removed entirely and all occurrences can be replaced with `IllegalArgumentException`. IAE has the advantage of being a standard exception, so that anyone who sees it will immediately understand what it means. The name is also more accurate (see [here](http://stackoverflow.com/q/156767) for the difference between "argument" and "parameter").

As far as I can tell, the only advantage of keeping `BadParameterException` is that it is more convenient to create the detail message. However, if that is really necessary, I think it would better to have a utility class/method which creates the message and throws the exception. Something like Guava's [`Preconditions`](http://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Preconditions.html) class.

Any thoughts, @dagnir?

**Comment by dagnir** [comment]
Yes I agree IllegalArgumentException works just as well in this case.
